### PR TITLE
refactor: Standardize names and add Arch Doc

### PR DESCRIPTION
**Pull request by carlaKC**
Before we run through another round of changes, let's standardize naming a little.

### CODE REVIEW

File: sim-lib/src/lib.rs [hunk 1]
```diff
@@ -131,8 +131,9 @@ pub trait LightningNode {
     async fn get_node_features(&mut self, node: PublicKey) -> Result<NodeFeatures, LightningError>;
 }
 
+/// SimulationEvent describes the set of actions that the simulator can excute on nodes that
```
Comment (okjodom) [inline sim-lib/src/lib.rs]: nit: incomplete comment?

Comment (carlaKC) [reply sim-lib/src/lib.rs]: Whoops, [pushed fixup](https://github.com/bitcoin-dev-project/sim-ln/compare/59a3887432b937aceffe9a0f2a537c3cab547a95..73591972c3e805f7e87605d9c7cb1ff118249eeb)

**Comment by okjodom** [comment]
Wish the arch doc included

- input of simulation events are input as activities. We can add to this when we have other activity types or define new ways of generating activities
- detail on how we support multiple-implementations of LN, and hinting at what it would take to add support for any other implementation

The rest looks good, i.e core arch of the simulator and how it execs events on nodes, tracks results and reports them

**Review by okjodom — APPROVED** [review]
naming lgtm

**Pull request merged**